test/controllers/crawlers_controller_test.rb

```diff
@@ -0,0 +1,59 @@
+require "test_helper"
+
+class CrawlersControllerTest < ActionController::TestCase
+  setup do
+    @crawler = crawlers(:one)
+  end
+
+  test "should get index" do
+    get :index
+    assert_response :success
+    assert_not_nil assigns(:crawlers)
+  end
+
+  test "should get new" do
+    get :new
+    assert_response :success
+  end
+
+  test "should create crawler" do
+    assert_difference("Crawler.count") do
+      post :create, crawler: { 
+                      desc: @crawler.desc, 
+                      notes: @crawler.notes, 
+                      title: @crawler.title, 
+                      url: @crawler.url 
+      }
+    end
+
+    assert_redirected_to crawler_path(assigns(:crawler))
+  end
+
+  test "should show crawler" do
+    get :show, id: @crawler
+    assert_response :success
+  end
+
+  test "should get edit" do
+    get :edit, id: @crawler
+    assert_response :success
+  end
+
+  test "should update crawler" do
+    patch :update, id: @crawler, crawler: { 
+                         desc: @crawler.desc, 
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.<br>Trailing whitespace detected.